mysql-test/t/select_outfile_fsync_debug.test

```diff
@@ -0,0 +1,63 @@
+disable_query_log;
+-- source include/test_outfile.inc
+# Server are started in "var/master-data", so "../../tmp" will be "var/tmp"
+eval set @tmpdir="../../tmp";
+enable_query_log;
+-- source include/have_outfile.inc
+-- source include/have_debug.inc
+
+# Save the initial number of concurrent sessions
+--source include/count_sessions.inc
+
+# Check for fsync's 
+--disable_warnings
+drop table if exists t2;
+--enable_warnings
+
+SET SESSION select_into_file_fsync_size=262144;
+SET GLOBAL debug = '+d,print_select_file_fsync_stats';
+
+let SEARCH_FILE = $MYSQLTEST_VARDIR/log/mysqld.1.err;
```

Comment:
Restart with custom error log path

Reply:
or use `assert_grep.inc` with `CURRENT_TEST: select_outfile_fsync_debug` e.g.
```
--let $assert_file= $MYSQLTEST_VARDIR/log/mysqld.1.err
--let $assert_only_after = CURRENT_TEST: binlog.binlog_cache_disk_full
--let $assert_count = 1
--let $assert_select = Disk is full writing .* \(OS errno 28 - No space left on device\). Waiting for someone to free space...
--let $assert_text = Found the expected error: No space left on device.
--source include/assert_grep.inc
```

Reply:
I think we discussed this before but I don't recall the outcome: is asset_only_after compatible with --repeat ?

Reply:
This was Satya's idea. It is compatible with `--repeat` and is has some advantages:
- no need to restart server (faster)
- everything is in logs (and a custom error log will be deleted).